File: lab01/cpp/tests/test_main.cpp
#include <catch2/catch_test_macros.hpp>
#include "../src/main.h"

TEST_CASE("Zuv daraalald oruulah") {
    int arr[] = {5,8,1,4,6};
    int n = 5;

    insertionSort(arr, n);

    REQUIRE(arr[0] == 1);
    REQUIRE(arr[1] == 4);
    REQUIRE(arr[2] == 5);
    REQUIRE(arr[3] == 6);
    REQUIRE(arr[4] == 8);
}

TEST_CASE("Zuv daraalaltai yed daraalal uurchlugduhgui") {
    int arr[] = {1, 2, 3, 4};
    insertionSort(arr, 4);

    REQUIRE(arr[0] == 1);
    REQUIRE(arr[1] == 2);
    REQUIRE(arr[2] == 3);
    REQUIRE(arr[3] == 4);
}

File: lab01/python/tests/test_main.py
import unittest
from src import main

class TestInsertionSort(unittest.TestCase):
    def test_insertion_sort(self):
        lst = [12,3,7,9,14,6,11,2]
        main.insertion_sort(lst, len(lst))
        self.assertEqual([2,3,6,7,9,11,12,14],lst)

if __name__ == "__main__":
    unittest.main()